mdal/mdal_data_model.hpp

```diff
@@ -38,6 +38,137 @@ namespace MDAL
 
   typedef std::vector< std::pair< std::string, std::string > > Metadata;
 
+  class DateTime;
+
+  class Duration
+  {
+    public:
+      enum Unit
+      {
+        milliseconds = 0,
+        seconds,
+        minutes,
+        hours,
+        days,
+        weeks
+      };
+
+      Duration();
+
+      ///TODO : implementation of operator, copy constructor, copy assignment, move operation
+
+      Duration( double duration, Unit unit );
+
+      double value( Unit unit ) const;
+
+    private:
+      Duration( int64_t ms );
+      int64_t mDuration; //in ms
+
+      friend class DateTime;
+  };
+
+  class DateTime
+  {
+    public:
+
+      //! Defaul constructor
+      DateTime(): mValid( false )
+      {}
+      //! Copy constructor
+      DateTime( const DateTime &other ): mJulianTime( other.mJulianTime )
+      {}
+
+      //static method returning DateTime
+
+      //! Create an instance from gregorian/julian calendar (or standart)
+      static DateTime fromStandartValue( int year, int month, int day, int hours = 0, int minutes = 0, double seconds = 0 );
+
+      //! Create an instance from julian day
+      static DateTime fromJulianDay( double julianDay )
+      {
+        return DateTime( int64_t( julianDay * 24 * 3600 * 1000 ) );
+      }
+
+      //! Create an default instance
+      static DateTime fromDefault()
+      {
+        return DateTime();
+      }
+
+      //! Returns a string with the date/time expressed in Greogrian/Julian calendar with ISO8601 format (local time zone)
+      std::string toStandartCalendarISO8601() const;
+
+      //! Returns the Julian day value
+      double toJulianDay() const;
+
+      //! Returns the Julain day value expressed with a string
+      std::string toJulianDayString() const
```

Comment:
why we need this?

Reply:
used in mdal_ascii_dat.cpp in `persist` method

Reply:
ah ok